lib/util.js

```diff
@@ -355,3 +355,55 @@ exports.stripHTML = (html) => {
     .replace(/<.*?>/gi, '')
     .trim();
 };
+
+/* Match begin and end braces of input JSON, return only json 
+ * it works with braced jsons only!
+ * @param {String} mixedJson
+ * @return {String}
+*/
+exports.cutAfterJSON = function(mixedJson) {
+  let counter = 0, pos = 0, isString, prevChar, open, close;
+  if (mixedJson[0] === '[') {
+    open = '[';
+    close = ']';
+  }
+  if (mixedJson[0] === '{') {
+    open = '{';
+    close = '}';
+  }
+  if (!open) {
+    throw new Error("can't cut unsupported JSON (need to begin with [ or { ) but got " + mixedJson[0]);
+  }
+  let result = '';
+  for (let character of mixedJson) {
+    result += character;
+    if (!isString && character === '"' && prevChar !== '\\') {
+      isString = true;
+      prevChar = character;
+      continue;
+    }
+    if (isString && character === '"' && prevChar !== '\\') {
+      isString = false;
+      prevChar = character;
+      continue
+    }
+    if (isString) {
+      prevChar = character;
+      continue;
+    }
+    if (prevChar === '\\') {
+      prevChar = character;
+      continue;
+    }
+    if (character === open) {
+      counter++;
+    }
+    if (character === close) {
+      counter--;
+    }
+    if (counter <= 0) {
+      return result;
+    }
+    prevChar = character;
+  }
```

Comment:
first of all - i love the approach
just wanna throw sth in here:
i'd personally prefer an original `for (let i = 0 ; xxxx)` loop, removing the need for the `character` and `prevChar` variable (since we can do array-like access on strings) as well as the `result` variable which could be replaced with a single `.substring`

Reply:
I changed to accumulated string because there was some problem with slice result on some urls (cutted more, than needed)

Reply:
prevChar is needed to correct parse entries like "\"" in strings, we need to ignore all braces inside string values...

Reply:
gave you an example in https://github.com/zba/node-ytdl-core/pull/1